sumpy/tools.py

```diff
@@ -756,9 +757,11 @@ def reduced_row_echelon_form(m):
             break
         pivot = nrows
         for k in range(index, nrows):
-            if mat[k, i] != 0 and pivot == nrows:
+            if isinstance(mat[k, i], sym.Basic) and not mat[k, i].is_number:
+                continue
```

Comment:
Wouldn't this consider arbitrary symbolic things zero? (Maybe add a comment to explain?

Reply:
Good catch. Will fix it.